app/src/main/java/us/shandian/giga/ui/adapter/MissionAdapter.java

```diff
@@ -678,9 +678,19 @@ private boolean handlePopupItem(@NonNull ViewHolderItem h, @NonNull MenuItem opt
                 shareFile(h.item.mission);
                 return true;
             case R.id.delete:
-                mDeleter.append(h.item.mission);
-                applyChanges();
-                checkMasterButtonsVisibility();
+                AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
+                builder.setMessage(R.string.delete_selected_file)
+                        .setPositiveButton(R.string.yes, (dialog, which) -> {
+                            mDeleter.append(h.item.mission);
+                            applyChanges();
+                            checkMasterButtonsVisibility();
+                        })
+                        .setNegativeButton(R.string.cancel, (dialog, which) -> {
+                            //nothing happens
+                        });
+                // Create the AlertDialog object and run it
+                builder.create();
+                builder.show();
```

Comment:
This improves the formatting, removes an unneeded variable, shows the name of the file being deleted in the dialog.
```suggestion
                new AlertDialog.Builder(mContext)
                        .setMessage(mContext.getString(R.string.delete_selected_file, h.item.mission.storage.getName()))
                        .setNegativeButton(R.string.cancel, null)
                        .setPositiveButton(R.string.yes, (dialog, which) -> {
                            mDeleter.append(h.item.mission);
                            applyChanges();
                            checkMasterButtonsVisibility();
                        })
                        .show();
```